src/plugins/instrument_fetch.js

```diff
@@ -0,0 +1,269 @@
+import * as opentracing from 'opentracing';
+
+// Capture the proxied values on script load (i.e. ASAP) in case there are
+// multiple layers of instrumentation.
+let proxiedFetch;
+if (typeof window === 'object' && typeof window.fetch !== 'undefined') {
+    proxiedFetch = window.fetch;
+}
+
+function getCookies() {
+    if (typeof document === 'undefined' || !document.cookie) {
+        return null;
+    }
+    let cookies = document.cookie.split(';');
+    let data = {};
+    let count = 0;
+    for (let i = 0; i < cookies.length; i++) {
+        let parts = cookies[i].split('=', 2);
+        if (parts.length === 2) {
+            let key = parts[0].replace(/^\s+/, '').replace(/\s+$/, '');
+            data[key] = decodeURIComponent(parts[1]);
+            try {
+                data[key] = JSON.parse(data[key]);
+            } catch (_ignored) { /* Ignored */ }
+            count++;
+        }
+    }
+    if (count > 0) {
+        return data;
+    }
+    return null;
+}
+
+// Normalize the getAllResponseHeaders output
+function getResponseHeaders(response) {
+    const result = {};
+    for (let pair of response.headers.entries()) {
+        result[pair[0]] = pair[1];
+    }
+    return result;
+}
+
+// Automatically create spans for all XMLHttpRequest objects.
+//
+// NOTE: this code currently works only with a single Tracer.
```

Comment:
Question: What would happen if there were multiple tracers?

Reply:
[this comment is inherited from the instrument_xhr plugin](https://github.com/lightstep/lightstep-tracer-javascript/blob/master/src/plugins/instrument_xhr.js#L52); I'm not really sure what happens if you have two tracers for either one.

Reply:
👍